File: .skills/openclaw-skills/skills/shixiangyu2/astronclaw-code-review/src/tools/report-generator.js
/**
 * 审查报告生成器
 * 生成Markdown、HTML格式的代码审查报告
 */

export class ReportGenerator {
  /**
   * 生成审查报告
   */
  static async execute(args, context) {
    const { 
      reviewResults, 
      format = 'markdown',
      includeDetails = true,
      options = {} 
    } = args;
    
    console.log(`📄 生成${format.toUpperCase()}格式审查报告...`);
    
    try {
      // 验证输入
      if (!reviewResults) {
        throw new Error('需要提供reviewResults参数');
      }
      
      let report;
      
      // 根据格式生成报告
      switch (format.toLowerCase()) {
        case 'markdown':
          report = ReportGenerator.generateMarkdownReport(reviewResults, includeDetails, options);
          break;
          
        case 'html':
          report = ReportGenerator.generateHTMLReport(reviewResults, includeDetails, options);
          break;
          
        case 'json':
          report = ReportGenerator.generateJSONReport(reviewResults, includeDetails, options);
          break;
          
        default:
          throw new Error(`不支持的格式: ${format}，支持: markdown, html, json`);
      }
      
      console.log(`✅ 报告生成完成: ${report.summary?.fileName || '未命名'}`);
      
      return {
        success: true,
        report,
        metadata: {
          format,
          generatedAt: new Date().toISOString(),
          size: ReportGenerator.calculateReportSize(report),
          includes: {
            details: includeDetails,
            recommendations: options.includeRecommendations !== false,
            suggestions: options.includeSuggestions !== false
          }
        }
      };
      
    } catch (error) {
      console.error('❌ 报告生成失败:', error.message);
      return {
        success: false,
        error: {
          message: error.message,
          code: 'REPORT_GENERATION_ERROR'
        }
      };
    }
  }
  
  /**
   * 生成Markdown报告
   */
  static generateMarkdownReport(results, includeDetails = true, options = {}) {
    const timestamp = new Date().toISOString();
    const fileName = results.summary?.filePath || 'code-review-report.md';
    
    let markdown = `# 代码审查报告\n\n`;
    
    // 报告头部
    markdown += `## 📋 报告概览\n\n`;
    markdown += `- **文件**: ${results.summary?.filePath || '未指定'}\n`;
    markdown += `- **生成时间**: ${timestamp}\n`;
    markdown += `- **总体评分**: ${results.summary?.overallScore || 0}/100\n`;
    markdown += `- **等级**: ${results.summary?.grade || 'N/A'}\n`;
    markdown += `- **代码行数**: ${results.summary?.linesOfCode || 0}\n`;
    markdown += `- **问题总数**: ${ReportGenerator.countTotalIssues(results) || 0}\n\n`;
    
    // 评分摘要
    markdown += `## 📊 评分摘要\n\n`;
    
    if (results.analysis) {
      markdown += `### 代码质量\n`;
      markdown += `- 评分: ${results.analysis.quality?.score || 0}/100\n`;
      markdown += `- 问题: ${results.analysis.quality?.issues?.length || 0} 个\n\n`;
      
      if (results.analysis.security?.enabled !== false) {
        markdown += `### 安全性\n`;
        markdown += `- 评分: ${results.analysis.security?.score || 0}/100\n`;
        markdown += `- 问题: ${results.analysis.security?.issues?.length || 0} 个\n\n`;
      }
      
      if (results.analysis.performance?.enabled !== false) {
        markdown += `### 性能\n`;
        markdown += `- 评分: ${results.analysis.performance?.score || 0}/100\n`;
        markdown += `- 问题: ${results.analysis.performance?.issues?.length || 0} 个\n\n`;
      }
    }
    
    // 关键问题
    const criticalIssues = ReportGenerator.getCriticalIssues(results);
    if (criticalIssues.length > 0) {
      markdown += `## 🚨 关键问题 (需要立即修复)\n\n`;
      
      criticalIssues.forEach((issue, index) => {
        markdown += `### ${index + 1}. ${issue.message}\n`;
        markdown += `- **位置**: 第 ${issue.line} 行\n`;
        markdown += `- **严重程度**: ${issue.severity}\n`;
        markdown += `- **建议**: ${issue.suggestion || '参考最佳实践'}\n\n`;
        
        if (issue.codeSnippet) {
          markdown += `\`\`\`\n${issue.codeSnippet}\n\`\`\`\n\n`;
        }
      });
    }
    
    // AI建议
    if (results.recommendations?.ai?.length > 0 && options.includeRecommendations !== false) {
      markdown += `## 🧠 AI智能建议\n\n`;
      
      results.recommendations.ai.forEach((rec, index) => {
        markdown += `### ${index + 1}. ${rec.title}\n`;
        markdown += `- **优先级**: ${rec.priority}\n`;
        markdown += `- **描述**: ${rec.description}\n`;
        
        if (rec.suggestion) {
          markdown += `- **具体建议**: ${rec.suggestion}\n`;
        }
        
        if (rec.codeExample) {
          markdown += `\n**代码示例**:\n\n\`\`\`\n${rec.codeExample}\n\`\`\`\n`;
        }
        
        markdown += '\n';
      });
    }
    
    // 详细问题列表（可选）
    if (includeDetails) {
      markdown += `## 🔍 详细问题列表\n\n`;
      
      ['quality', 'security', 'performance'].forEach(category => {
        const categoryData = results.analysis?.[category];
        if (categoryData && categoryData.issues?.length > 0) {
          markdown += `### ${ReportGenerator.getCategoryName(category)}\n\n`;
          
          categoryData.issues.forEach((issue, idx) => {
            markdown += `${idx + 1}. **${issue.message}**\n`;
            markdown += `   - 行号: ${issue.line || 'N/A'}\n`;
            markdown += `   - 严重程度: ${issue.severity || 'info'}\n`;
            if (issue.suggestion) {
              markdown += `   - 建议: ${issue.suggestion}\n`;
            }
            markdown += '\n';
          });
        }
      });
    }
    
    // 改进建议
    markdown += `## 🛠️ 改进计划\n\n`;
    
    if (results.recommendations?.priority?.length > 0) {
      markdown += `### 优先级排序\n\n`;
      
      results.recommendations.priority.forEach((item, index) => {
        markdown += `${index + 1}. **${item.priority.toUpperCase()}**: ${item.message || '未指定'}\n`;
        if (item.suggestion) {
          markdown += `   - 建议: ${item.suggestion}\n`;
        }
        markdown += '\n';
      });
    }
    
    // 最佳实践
    markdown += `## 📚 最佳实践参考\n\n`;
    markdown += `1. **代码质量**: 遵循编码规范，保持代码简洁清晰\n`;
    markdown += `2. **安全性**: 验证所有输入，避免硬编码敏感信息\n`;
    markdown += `3. **性能**: 避免不必要的计算，优化关键路径\n`;
    markdown += `4. **可维护性**: 添加适当注释，保持函数单一职责\n`;
    markdown += `5. **可测试性**: 编写单元测试，确保代码质量\n\n`;
    
    // 报告脚注
    markdown += `---\n\n`;
    markdown += `*本报告由 AI Code Review Assistant 生成*\n`;
    markdown += `*生成工具: AstronClaw Skill - Code Review Assistant*\n`;
    markdown += `*报告版本: 1.0.0*\n`;
    
    return {
      content: markdown,
      format: 'markdown',
      fileName: fileName.endsWith('.md') ? fileName : `${fileName}.md`,
      size: markdown.length,
      sections: ReportGenerator.extractSections(markdown)
    };
  }
  
  /**
   * 生成HTML报告
   */
  static generateHTMLReport(results, includeDetails = true, options = {}) {
    const markdownReport = ReportGenerator.generateMarkdownReport(results, includeDetails, options);
    const markdown = markdownReport.content;
    
    // 简单的Markdown转HTML（实际实现应使用更完善的转换器）
    let html = `<!DOCTYPE html>
<html lang="zh-CN">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>代码审查报告 - ${results.summary?.filePath || '未命名'}</title>
    <style>
        body { font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, sans-serif; line-height: 1.6; max-width: 1200px; margin: 0 auto; padding: 20px; color: #333; }
        h1, h2, h3, h4 { color: #2c3e50; margin-top: 1.5em; }
        h1 { border-bottom: 2px solid #3498db; padding-bottom: 10px; }
        h2 { border-bottom: 1px solid #ddd; padding-bottom: 5px; }
        .critical { color: #e74c3c; font-weight: bold; }
        .warning { color: #f39c12; }
        .info { color: #3498db; }
        .success { color: #27ae60; }
        .score { font-size: 2em; font-weight: bold; margin: 10px 0; }
        .score-excellent { color: #27ae60; }
        .score-good { color: #2ecc71; }
        .score-average { color: #f39c12; }
        .score-poor { color: #e74c3c; }
        table { border-collapse: collapse; width: 100%; margin: 20px 0; }
        th, td { border: 1px solid #ddd; padding: 12px; text-align: left; }
        th { background-color: #f2f2f2; font-weight: bold; }
        tr:nth-child(even) { background-color: #f9f9f9; }
        .code-block { background-color: #f5f5f5; padding: 15px; border-radius: 5px; font-family: 'Courier New', monospace; overflow-x: auto; }
        .summary-card { background: #f8f9fa; border-left: 4px solid #3498db; padding: 15px; margin: 20px 0; border-radius: 4px; }
        .recommendation { background: #e8f4fc; border-left: 4px solid #3498db; padding: 15px; margin: 15px 0; border-radius: 4px; }
        .footer { margin-top: 40px; padding-top: 20px; border-top: 1px solid #ddd; color: #7f8c8d; font-size: 0.9em; }
        @media print { body { max-width: none; } .no-print { display: none; } }
    </style>
</head>
<body>
    <h1>📋 代码审查报告</h1>
    
    <div class="summary-card">
        <h2>报告概览</h2>
        <p><strong>文件:</strong> ${results.summary?.filePath || '未指定'}</p>
        <p><strong>生成时间:</strong> ${new Date().toISOString()}</p>
        <p><strong>总体评分:</strong> 
            <span class="score ${ReportGenerator.getScoreClass(results.summary?.overallScore || 0)}">
                ${results.summary?.overallScore || 0}/100
            </span>
        </p>
        <p><strong>等级:</strong> ${results.summary?.grade || 'N/A'}</p>
        <p><strong>代码行数:</strong> ${results.summary?.linesOfCode || 0}</p>
        <p><strong>问题总数:</strong> ${ReportGenerator.countTotalIssues(results) || 0}</p>
    </div>
`;
    
    // 添加评分摘要表格
    html += `
    <h2>📊 评分摘要</h2>
    <table>
        <thead>
            <tr>
                <th>类别</th>
                <th>评分</th>
                <th>问题数</th>
                <th>状态</th>
            </tr>
        </thead>
        <tbody>`;
    
    if (results.analysis) {
      html += `
            <tr>
                <td><strong>代码质量</strong></td>
                <td>${results.analysis.quality?.score || 0}/100</td>
                <td>${results.analysis.quality?.issues?.length || 0}</td>
                <td class="${ReportGenerator.getStatusClass(results.analysis.quality?.score || 0)}">
                    ${ReportGenerator.getStatusText(results.analysis.quality?.score || 0)}
                </td>
            </tr>`;
      
      if (results.analysis.security?.enabled !== false) {
        html += `
            <tr>
                <td><strong>安全性</strong></td>
                <td>${results.analysis.security?.score || 0}/100</td>
                <td>${results.analysis.security?.issues?.length || 0}</td>
                <td class="${this.getStatusClass(results.analysis.security?.score || 0)}">
                    ${this.getStatusText(results.analysis.security?.score || 0)}
                </td>
            </tr>`;
      }
      
      if (results.analysis.performance?.enabled !== false) {
        html += `
            <tr>
                <td><strong>性能</strong></td>
                <td>${results.analysis.performance?.score || 0}/100</td>
                <td>${results.analysis.performance?.issues?.length || 0}</td>
                <td class="${this.getStatusClass(results.analysis.performance?.score || 0)}">
                    ${this.getStatusText(results.analysis.performance?.score || 0)}
                </td>
            </tr>`;
      }
    }
    
    html += `
        </tbody>
    </table>`;
    
    // 添加关键问题部分
    const criticalIssues = this.getCriticalIssues(results);
    if (criticalIssues.length > 0) {
      html += `
    <h2 class="critical">🚨 关键问题 (需要立即修复)</h2>`;
      
      criticalIssues.forEach((issue, index) => {
        html += `
    <div class="recommendation">
        <h3>${index + 1}. ${issue.message}</h3>
        <p><strong>位置:</strong> 第 ${issue.line} 行</p>
        <p><strong>严重程度:</strong> <span class="critical">${issue.severity}</span></p>
        <p><strong>建议:</strong> ${issue.suggestion || '参考最佳实践'}</p>`;
        
        if (issue.codeSnippet) {
          html += `
        <div class="code-block">
            <pre>${issue.codeSnippet}</pre>
        </div>`;
        }
        
        html += `
    </div>`;
      });
    }
    
    // 报告脚注
    html += `
    <div class="footer">
        <p>本报告由 <strong>AI Code Review Assistant</strong> 生成</p>
        <p>生成工具: AstronClaw Skill - Code Review Assistant v1.0.0</p>
        <p class="no-print">报告ID: ${Date.now().toString(36)}</p>
    </div>
</body>
</html>`;
    
    return {
      content: html,
      format: 'html',
      fileName: results.summary?.filePath ? 
        `${results.summary.filePath.replace(/\.[^/.]+$/, "")}-report.html` : 
        'code-review-report.html',
      size: html.length
    };
  }
  
  /**
   * 生成JSON报告
   */
  static generateJSONReport(results, includeDetails = true, options = {}) {
    const report = {
      metadata: {
        generator: 'AI Code Review Assistant',
        version: '1.0.0',
        generatedAt: new Date().toISOString(),
        format: 'json'
      },
      summary: results.summary || {},
      analysis: includeDetails ? results.analysis : undefined,
      recommendations: options.includeRecommendations !== false ? results.recommendations : undefined,
      statistics: {
        totalIssues: this.countTotalIssues(results),
        criticalIssues: this.getCriticalIssues(results).length,
        analysisTime: results.metadata?.analysisTime
      }
    };
    
    return {
      content: JSON.stringify(report, null, 2),
      format: 'json',
      fileName: results.summary?.filePath ? 
        `${results.summary.filePath.replace(/\.[^/.]+$/, "")}-report.json` : 
        'code-review-report.json',
      size: JSON.stringify(report).length
    };
  }
  
  /**
   * 统计总问题数
   */
  static countTotalIssues(results) {
    let total = 0;
    
    if (results.analysis) {
      ['quality', 'security', 'performance'].forEach(category => {
        if (results.analysis[category]?.issues) {
          total += results.analysis[category].issues.length;
        }
      });
    }
    
    return total;
  }
  
  /**
   * 获取关键问题
   */
  static getCriticalIssues(results) {
    const criticalIssues = [];
    
    if (results.analysis) {
      ['quality', 'security', 'performance'].forEach(category => {
        const issues = results.analysis[category]?.issues || [];
        issues.forEach(issue => {
          if (issue.severity === 'critical' || issue.severity === 'high') {
            criticalIssues.push({
              ...issue,
              category
            });
          }
        });
      });
    }
    
    return criticalIssues;
  }
  
  /**
   * 获取类别名称
   */
  static getCategoryName(category) {
    const names = {
      quality: '代码质量',
      security: '安全性',
      performance: '性能'
    };
    
    return names[category] || category;
  }
  
  /**
   * 提取章节
   */
  static extractSections(markdown) {
    const sections = [];
    const lines = markdown.split('\n');
    
    lines.forEach(line => {
      if (line.startsWith('## ')) {
        sections.push(line.replace('## ', '').trim());
      }
    });
    
    return sections;
  }
  
  /**
   * 计算报告大小
   */
  static calculateReportSize(report) {
    if (typeof report.content === 'string') {
      return report.content.length;
    }
    return 0;
  }
  
  /**
   * 获取评分CSS类
   */
  static getScoreClass(score) {
    if (score >= 90) return 'score-excellent';
    if (score >= 80) return 'score-good';
    if (score >= 70) return 'score-average';
    return 'score-poor';
  }
  
  /**
   * 获取状态CSS类
   */
  static getStatusClass(score) {
    if (score >= 80) return 'success';
    if (score >= 60) return 'warning';
    return 'critical';
  }
  
  /**
   * 获取状态文本
   */
  static getStatusText(score) {
    if (score >= 90) return '优秀';
    if (score >= 80) return '良好';
    if (score >= 70) return '中等';
    if (score >= 60) return '及格';
    return '需要改进';
  }
}

File: .skills/openclaw-skills/skills/sjj2026/lightweight-autoresearch/scripts/report_generator.py
#!/usr/bin/env python3
"""
报告生成器 - 生成 HTML 可视化报告
"""

import os
import json
from datetime import datetime
from pathlib import Path
from typing import Dict, List

class ReportGenerator:
    """报告生成器"""
    
    def __init__(self, skill_path: str):
        self.skill_path = Path(skill_path)
        self.report_dir = self.skill_path / "reports"
        
    def generate(self, evaluation_results: Dict, code_results: Dict = None,
                 security_results: Dict = None, performance_results: Dict = None) -> str:
        """生成完整报告"""
        
        # 创建报告目录
        self.report_dir.mkdir(exist_ok=True)
        
        # 生成报告
        report_time = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
        report_file = self.report_dir / f"report_{report_time}.html"
        
        # 构建 HTML
        html = self._build_html(
            evaluation_results,
            code_results,
            security_results,
            performance_results,
            report_time
        )
        
        # 写入文件
        report_file.write_text(html, encoding='utf-8')
        
        return str(report_file)
    
    def _build_html(self, evaluation: Dict, code: Dict, 
                    security: Dict, performance: Dict, report_time: str) -> str:
        """构建 HTML 报告"""
        
        total_score = evaluation.get("total_score", 0)
        
        # 计算综合分数
        if code and security and performance:
            comprehensive_score = (
                evaluation["total_score"] * 0.3 +
                code["quality_score"] * 0.25 +
                security["security_score"] * 0.25 +
                performance["performance_score"] * 0.2
            )
        else:
            comprehensive_score = total_score
        
        html = f'''<!DOCTYPE html>
<html lang="zh-CN">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>技能评估报告 - {self.skill_path.name}</title>
    <style>
        * {{
            margin: 0;
            padding: 0;
            box-sizing: border-box;
        }}
        
        body {{
            font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, sans-serif;
            background: linear-gradient(135deg, #667eea 0%, #764ba2 100%);
            padding: 20px;
            color: #333;
        }}
        
        .container {{
            max-width: 1200px;
            margin: 0 auto;
            background: white;
            border-radius: 20px;
            box-shadow: 0 20px 60px rgba(0,0,0,0.3);
            overflow: hidden;
        }}
        
        .header {{
            background: linear-gradient(135deg, #667eea 0%, #764ba2 100%);
            color: white;
            padding: 40px;
            text-align: center;
        }}
        
        .header h1 {{
            font-size: 2.5em;
            margin-bottom: 10px;
        }}
        
        .header p {{
            opacity: 0.9;
            font-size: 1.1em;
        }}
        
        .score-section {{
            padding: 40px;
            background: #f8f9fa;
        }}
        
        .score-card {{
            background: white;
            border-radius: 15px;
            padding: 30px;
            text-align: center;
            box-shadow: 0 5px 15px rgba(0,0,0,0.1);
            margin-bottom: 20px;
        }}
        
        .score-circle {{
            width: 200px;
            height: 200px;
            border-radius: 50%;
            margin: 0 auto 20px;
            display: flex;
            align-items: center;
            justify-content: center;
            font-size: 4em;
            font-weight: bold;
            color: white;
            background: {self._get_score_color(comprehensive_score)};
            box-shadow: 0 10px 30px rgba(0,0,0,0.2);
        }}
        
        .score-label {{
            font-size: 1.5em;
            color: #666;
            margin-bottom: 10px;
        }}
        
        .section {{
            padding: 40px;
            border-bottom: 1px solid #eee;
        }}
        
        .section:last-child {{
            border-bottom: none;
        }}
        
        .section h2 {{
            font-size: 1.8em;
            margin-bottom: 20px;
            color: #333;
            display: flex;
            align-items: center;
        }}
        
        .section h2:before {{
            content: "";
            width: 5px;
            height: 30px;
            background: #667eea;
            margin-right: 15px;
            border-radius: 3px;
        }}
        
        .metrics-grid {{
            display: grid;
            grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
            gap: 20px;
            margin-top: 20px;
        }}
        
        .metric-card {{
            background: #f8f9fa;
            border-radius: 10px;
            padding: 20px;
            border-left: 4px solid #667eea;
        }}
        
        .metric-title {{
            font-size: 0.9em;
            color: #666;
            margin-bottom: 10px;
        }}
        
        .metric-value {{
            font-size: 2em;
            font-weight: bold;
            color: #333;
        }}
        
        .issues-list {{
            margin-top: 20px;
        }}
        
        .issue {{
            background: #fff3cd;
            border-left: 4px solid #ffc107;
            padding: 15px;
            margin-bottom: 10px;
            border-radius: 5px;
        }}
        
        .issue.error {{
            background: #f8d7da;
            border-left-color: #dc3545;
        }}
        
        .issue.warning {{
            background: #fff3cd;
            border-left-color: #ffc107;
        }}
        
        .issue.success {{
            background: #d4edda;
            border-left-color: #28a745;
        }}
        
        .progress-bar {{
            width: 100%;
            height: 10px;
            background: #e9ecef;
            border-radius: 5px;
            overflow: hidden;
            margin-top: 10px;
        }}
        
        .progress-fill {{
            height: 100%;
            background: #667eea;
            transition: width 0.3s ease;
        }}
        
        .footer {{
            background: #f8f9fa;
            padding: 20px;
            text-align: center;
            color: #666;
            font-size: 0.9em;
        }}
        
        @media print {{
            body {{
                background: white;
                padding: 0;
            }}
            
            .container {{
                box-shadow: none;
            }}
        }}
    </style>
</head>
<body>
    <div class="container">
        <div class="header">
            <h1>🎯 技能评估报告</h1>
            <p>{self.skill_path.name}</p>
            <p style="margin-top: 10px; opacity: 0.8;">生成时间: {report_time}</p>
        </div>
        
        <div class="score-section">
            <div class="score-card">
                <div class="score-circle">
                    {comprehensive_score:.0f}
                </div>
                <div class="score-label">综合评分</div>
                <p style="color: #666; margin-top: 10px;">
                    {self._get_score_description(comprehensive_score)}
                </p>
            </div>
        </div>
        
        <!-- 基础评估 -->
        {self._build_evaluation_section(evaluation)}
        
        <!-- 代码质量 -->
        {self._build_code_section(code) if code else ""}
        
        <!-- 安全检查 -->
        {self._build_security_section(security) if security else ""}
        
        <!-- 性能测试 -->
        {self._build_performance_section(performance) if performance else ""}
        
        <!-- 改进建议 -->
        {self._build_suggestions_section(evaluation, code, security)}
        
        <div class="footer">
            <p>🦞 轻量级自主优化技能 - 自动生成报告</p>
        </div>
    </div>
</body>
</html>'''
        
        return html
    
    def _build_evaluation_section(self, evaluation: Dict) -> str:
        """构建评估部分"""
        details = evaluation.get("details", {})
        
        metrics_html = ""
        for category, data in details.items():
            score = data.get("score", 0)
            max_score = data.get("max_score", 100)
            percentage = (score / max_score * 100) if max_score > 0 else 0
            
            metrics_html += f'''
            <div class="metric-card">
                <div class="metric-title">{self._get_category_name(category)}</div>
                <div class="metric-value">{score}/{max_score}</div>
                <div class="progress-bar">
                    <div class="progress-fill" style="width: {percentage}%"></div>
                </div>
            </div>'''
        
        return f'''
        <div class="section">
            <h2>📊 基础评估</h2>
            <div class="metrics-grid">
                {metrics_html}
            </div>
        </div>'''
    
    def _build_code_section(self, code: Dict) -> str:
        """构建代码质量部分"""
        quality_score = code.get("quality_score", 0)
        issues = code.get("issues", [])
        
        issues_html = ""
        for issue in issues[:5]:  # 只显示前5个
            issue_class = issue.get("type", "warning")
            message = issue.get("message", "")
            file_name = issue.get("file", "")
            
            issues_html += f'''
            <div class="issue {issue_class}">
                <strong>{file_name}</strong>: {message}
            </div>'''
        
        return f'''
        <div class="section">
            <h2>🔧 代码质量</h2>
            <div class="metrics-grid">
                <div class="metric-card">
                    <div class="metric-title">代码质量分数</div>
                    <div class="metric-value">{quality_score}/100</div>
                </div>
            </div>
            <div class="issues-list">
                {issues_html}
            </div>
        </div>'''
    
    def _build_security_section(self, security: Dict) -> str:
        """构建安全检查部分"""
        security_score = security.get("security_score", 0)
        issues = security.get("issues", [])
        
        issues_html = ""
        for issue in issues[:5]:
            severity = issue.get("severity", "low")
            message = issue.get("message", "")
            file_name = issue.get("file", "")
            
            issue_class = "error" if severity == "high" else "warning"
            
            issues_html += f'''
            <div class="issue {issue_class}">
                <strong>{file_name}</strong>: {message}
            </div>'''
        
        if not issues:
            issues_html = '<div class="issue success">✅ 未发现安全问题</div>'
        
        return f'''
        <div class="section">
            <h2>🔒 安全检查</h2>
            <div class="metrics-grid">
                <div class="metric-card">
                    <div class="metric-title">安全分数</div>
                    <div class="metric-value">{security_score}/100</div>
                </div>
            </div>
            <div class="issues-list">
                {issues_html}
            </div>
        </div>'''
    
    def _build_performance_section(self, performance: Dict) -> str:
        """构建性能测试部分"""
        performance_score = performance.get("performance_score", 0)
        metrics = performance.get("metrics", {})
        
        metrics_html = ""
        
        if "import_time" in metrics and metrics["import_time"]:
            metrics_html += f'''
            <div class="metric-card">
                <div class="metric-title">导入时间</div>
                <div class="metric-value">{metrics["import_time"]:.3f}s</div>
            </div>'''
        
        if "execution_time" in metrics and metrics["execution_time"]:
            metrics_html += f'''
            <div class="metric-card">
                <div class="metric-title">执行时间</div>
                <div class="metric-value">{metrics["execution_time"]:.2f}s</div>
            </div>'''
        
        if "estimated_memory_mb" in metrics:
            metrics_html += f'''
            <div class="metric-card">
                <div class="metric-title">估算内存</div>
                <div class="metric-value">{metrics["estimated_memory_mb"]:.2f}MB</div>
            </div>'''
        
        return f'''
        <div class="section">
            <h2>⚡ 性能测试</h2>
            <div class="metrics-grid">
                <div class="metric-card">
                    <div class="metric-title">性能分数</div>
                    <div class="metric-value">{performance_score}/100</div>
                </div>
                {metrics_html}
            </div>
        </div>'''
    
    def _build_suggestions_section(self, evaluation: Dict, code: Dict, security: Dict) -> str:
        """构建改进建议部分"""
        suggestions = []
        
        # 从评估中提取建议
        if evaluation.get("suggestions"):
            suggestions.extend(evaluation["suggestions"])
        
        # 从代码质量中提取建议
        if code and code.get("suggestions"):
            suggestions.extend(code["suggestions"])
        
        # 去重
        suggestions = list(set(suggestions))[:10]  # 最多10条
        
        suggestions_html = ""
        for i, suggestion in enumerate(suggestions, 1):
            suggestions_html += f'''
            <div class="issue success">
                <strong>{i}.</strong> {suggestion}
            </div>'''
        
        if not suggestions:
            suggestions_html = '<div class="issue success">✅ 技能质量优秀，无需改进</div>'
        
        return f'''
        <div class="section">
            <h2>💡 改进建议</h2>
            <div class="issues-list">
                {suggestions_html}
            </div>
        </div>'''
    
    def _get_score_color(self, score: float) -> str:
        """根据分数获取颜色"""
        if score >= 80:
            return "linear-gradient(135deg, #00c851 0%, #00a844 100%)"
        elif score >= 60:
            return "linear-gradient(135deg, #ffbb33 0%, #ff8800 100%)"
        else:
            return "linear-gradient(135deg, #ff4444 0%, #cc0000 100%)"
    
    def _get_score_description(self, score: float) -> str:
        """根据分数获取描述"""
        if score >= 80:
            return "✅ 技能质量优秀"
        elif score >= 60:
            return "⚠️ 技能质量良好，有改进空间"
        else:
            return "❌ 技能质量需要改进"
    
    def _get_category_name(self, category: str) -> str:
        """获取分类名称"""
        names = {
            "skill_md": "SKILL.md 完整性",
            "structure": "目录结构",
            "tests": "测试覆盖",
            "documentation": "文档质量",
            "python": "Python 代码",
            "shell": "Shell 脚本",
            "config": "配置文件"
        }
        return names.get(category, category)


def generate_report(skill_path: str, evaluation: Dict, code: Dict = None,
                   security: Dict = None, performance: Dict = None) -> str:
    """生成报告的便捷函数"""
    generator = ReportGenerator(skill_path)
    return generator.generate(evaluation, code, security, performance)
